beacon-light-client/plonky2/circuits_executables/bin/balance_verification_circuit_data_generation.rs

```diff
@@ -59,6 +59,11 @@ pub async fn async_main() -> Result<()> {
         _phantom: PhantomData::<PoseidonGoldilocksConfig>,
     };
 
+    if level != None && level.unwrap() > 37 {
+        eprintln!("\x1b[31mError: Supplied level {} is larger than the maximum allowed level 38\x1b[0m", level.unwrap());
+        process::exit(1);
```

Comment:
This function returns Result, why not use return Error instead of exit?

Reply:
I've made several attempts to make use of Err, however all of them give issues regarding the argument given to Err. The Result is Result<()>, and I'm not quite sure how to make use of Err, with it.